homework/index.js

```diff
@@ -23,24 +10,151 @@
+    createAndAppend('h1', root, { text: error.message, class: 'alert-error' });
+  }
+
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status >= 200 && xhr.status <= 299) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+  function showContributions(urlText) {
+    const infoRight = createAndAppend('div', infoContainer, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+
+    function renderContributions(contributors) {
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+        });
+        imgDiv.firstChild.style.maxHeight = '50px';
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
+      if (infoRight.firstElementChild === null) {
+        createAndAppend('p', infoRight, {
+          class: 'no-content',
+          text: 'There are no contributions yet.',
+        });
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    }
+
+    fetchJSON(urlText)
+      .then(goodResult => renderContributions(goodResult))
+      .catch(somethingWrong => renderError(somethingWrong));
+  }
+
+  function showRepoInfo(wantedRepo) {
+    const infoLeft = createAndAppend('div', infoContainer, {
+      class: 'info-left',
+      text: 'Repository information:',
     });
+    const table = createAndAppend('table', infoLeft);
+    const tbody = createAndAppend('tbody', table);
+    addRow(tbody, `Name`, wantedRepo.name);
+    addRow(tbody, `Description`, wantedRepo.description);
+    addRow(tbody, `Forks`, wantedRepo.forks);
+    addRow(tbody, `Last update`, new Date(wantedRepo.updated_at).toLocaleString());
+    const toBeLink = tbody.firstChild.childNodes[1];
+    toBeLink.innerHTML = `<a href="${wantedRepo.html_url}" target="_blank">${wantedRepo.name}</a>`;
```

Comment:
This is ugly and inconsistent with the rest of the code. You can use `createAndAppend` instead:

```js
createAndAppend('a', firstRow, { href: ..., target: ..., text: ... });
``

Reply:
I need to embed the anchor tag in the cell, exactly the second cell. So I have just wrote like: 
`createAndAppend('a', firstRow.childNodes[1], {
      text: ....,
      href: .....,
      target: ....',
    });`
Is that good? or doe you mean that I do not need to use childNodes[]?

Reply:
Ah yes, the easiest is to use firstRow.lastChild